tsconfig.json

```diff
@@ -9,12 +9,13 @@
     "strictNullChecks": true,
     "noImplicitAny": false,
     "experimentalDecorators": true,
+    "emitDecoratorMetadata": true,
```

Comment:
This might have caused some issues regarding the types above. I don't actually think we need this. This is because we don't actually use the `reflect-metadata` module at all.

Reply:
https://www.typescriptlang.org/tsconfig#emitDecoratorMetadata

Reply:
I'm removing this. @tegefaulkes please re-run the eslint later, to get rid of all the other file changes. This should focus precisely on the `tasks` and `contexts` system.

Reply:
Pushed linting changes.